cmd/vspd/spentticket.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2023 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.package main
+
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/decred/dcrd/blockchain/stake/v5"
+	"github.com/decred/dcrd/rpc/jsonrpc/types/v4"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/vspd/database"
+)
+
+type spentTicket struct {
+	dbTicket     database.Ticket
+	expiryHeight int64
+	heightSpent  int64
+	spendingTx   *wire.MsgTx
+}
+
+func (s *spentTicket) voted() bool {
+	return stake.IsSSGen(s.spendingTx)
+}
+
+// findSpentTickets attempts to find transactions that vote/revoke the provided
+// tickets by matching the payment script of the tickets commitment address
+// against the block filters of the mainchain blocks between the provided start
+// block and the current best block. Returns any found spent tickets and the
+// height of the most recent scanned block.
+func (v *vspd) findSpentTickets(toCheck database.TicketList, startHeight int64) (map[string]spentTicket, int64, error) {
+	params := v.cfg.netParams
+
+	dcrdClient, _, err := v.dcrd.Client()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	bestBlock, err := dcrdClient.GetBestBlockHeader()
```

Comment:
This isn't really performance critical, so it's not a big deal, but given the entire point of this is to find the most recent height, could just call [getblockcount](https://github.com/decred/dcrd/blob/0a35adc02120805f5185cc0a53eb2674711e3013/docs/json_rpc_api.mediawiki#user-content-getblockcount) rather than getting a header and decoding it.